### ISSUE TITLE
checkpoint issue

### ISSUE THREAD

**Issue by cyberinQ** [Edoar-do/HuBERT-ECG#7]
As soon I change the model from _large to _base, I start running into the checkpoint issue. What could I be doing wrong?

# --- Model loading ---
# Path to pretrained model checkpoint
model_path = "/Users/suniljain/01/jain-ecgfm/[hubert_base_cardio_learning.pt](http://hubert_base_cardio_learning.pt/)"  # Update with   path

# Load the saved model checkpoint
checkpoint = torch.load(model_path, map_location=torch.device('cpu'), weights_only=False)

# Extract the model configuration from the checkpoint
config = HuBERTECGConfig(**checkpoint['model_config'].to_dict())

# Create a HuBERTECG model instance with the extracted configuration
pretrained_hubert = HuBERTECG(config)

# --- Get the number of labels from the model ---
num_labels_from_model = checkpoint['finetuning_vocab_size']
print(f"Number of labels from model: {num_labels_from_model}")

# Create a HuBERTForECGClassification model instance
model = HuBERTForECGClassification(
    pretrained_hubert,
    num_labels=num_labels_from_model,  # Use the value from the checkpoint
    classifier_hidden_size=None if checkpoint['linear'] or checkpoint['use_label_embedding'] else checkpoint['model_state_dict'].values()[-2].size(-1),
    use_label_embedding=checkpoint['use_label_embedding']
)

# Load the model weights from the checkpoint
model.load_state_dict(checkpoint['model_state_dict'])

# Set the model to evaluation mode
model.eval()

print("Model loaded and set to evaluation mode.")

**Comment by cyberinQ** [comment]
Resolved. Directory error.
